bin/cli.js

```diff
@@ -1,15 +1,297 @@
 #!/usr/bin/env node
 
 const fs = require('fs-extra');
-const argv = require('minimist')(process.argv.slice(2));
-const childProcess = require('child_process');
-const inquirer = require('inquirer');
-const prompt = inquirer.createPromptModule();
-const util = require('util');
-const exec = util.promisify(childProcess.exec);
-const spawn = util.promisify(childProcess.spawn);
-const fork = util.promisify(childProcess.fork);
-
-let command = argv._[0];
-
-// TODO: Implement
+const ldposClient = require('ldpos-client');
+const { CliInterface } = require('@maartennnn/cli-builder');
+const actions = require('../lib/actions');
+const { FULL_CONFIG_PATH } = require('../lib/constants');
+
+const cli = new CliInterface({
+  interactive: true,
+  helpFooter: '',
+  helpHeader: 'This is shown in the header',
+  exceptions: ['clean'],
+  actions,
+});
+
+(async () => {
+  let config = {};
+  let client;
+
+  // If command is an ip change config to that IP
+  if (
+    cli.argv._[0] &&
+    cli.argv._[0].includes('.') &&
+    cli.argv._[0].split('.').length === 4
+  ) {
+    const hostname = cli.argv._[0].split(':')[0];
+    const port = cli.argv._[0].split(':')[1] || 7001;
+
+    // Take out the IP if provided
+    cli.argv._ = cli.argv._.slice(1);
+
+    config = { hostname, port };
+  }
+
+  if (
+    !cli.options.exceptions
+      .map((f) => cli.argv.hasOwnProperty(f) || cli.argv._.includes(f))
+      .includes(true)
+  ) {
+    if (!config.hostname) {
+      if (await fs.pathExists(FULL_CONFIG_PATH)) {
+        config = require(FULL_CONFIG_PATH);
+      } else {
+        // prettier-ignore
+        config = {
+          ...config,
+          hostname: await cli.promptInput('Server IP:'),
+          port: await cli.promptInput('Port: (Default: 7001)') || 7001,
+        };
+
+        if(config.hostname === '') return cli.errorLog('Hostname needs to be provided')
+
+        const save = ['Y', 'y'].includes(
+          await cli.promptInput(`Save in your home dir? (Y/n)`)
+        );
+
+        if (save)
+          await fs.outputFile(
+            FULL_CONFIG_PATH,
+            JSON.stringify(config, null, 2)
+          );
+      }
+    }
+
+    if (!config.networkSymbol) {
+      config = {
+        ...config,
+        networkSymbol:
+          (await cli.promptInput('Network symbol: (Default: ldpos)')) ||
+          'ldpos',
+      };
+    }
+
+    // Get passphrase of the wallet
+    config.passphrase =
+      (await cli.promptInput('Passphrase:', true));
+
+    if(config.hostname === '') return cli.errorLog('Passphrase needs to be provided')
+
+    client = ldposClient.createClient(config);
+
+    try {
+      await client.connect({
+        passphrase: config.passphrase,
+      });
+
+      cli.options.bindActionArgs = [client];
+    } catch (e) {
+      cli.errorLog("Can't connect to socket");
+    }
+  }
+
+  const ldposAction = async (clientKey, message, action = null) => {
```

Comment:
Not sure about the name of the last parameter here. If it's the argument for the client, maybe it should be called `arg` or `args` plural? Also what if the method had multiple arguments (not just one)? Should it be an array?

Reply:
resolved in 8688b5e

Reply:
@jondubois I changed it to `arg`, momentarily it's only used for `sig`, `multisig` and `forging`.

Reply:
resolved in b2d1cd3